src/app/energy-summary-list/model/calculateNetMetering.ts

```
@@ -0,0 +1,105 @@
+    battery: Battery,
+    prosumentType: ProsumerType = ProsumerType.netMetering70
+): EnergyData {
+    let batteryCharge = 0; // aktualny stan naładowania baterii w kWh
+    let selfConsumption = 0; // autokonsumpcja w kWh
+    let fedIntoGrid = 0; // wyslane do sieci
+
+  const gridEnergyMultiplayer = prosumentType === ProsumerType.netMetering70 ? 0.7 : 0.8;
+
+
+    const initializeTariffData = (zones: TariffZone[]): TariffEnergyAmountData => ({
+        amount: zones.reduce((acc, zone) => ({...acc, [zone]: 0}), {[TariffZone.Day]: 0}),
```

Biome

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)